### PR TITLE
task_1 solution

### PR DESCRIPTION
**Pull request by MorozVladislav**

### CODE REVIEW

File: solutions/task_1.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def func_2():
+    return func_1
+
+
+if __name__ == '__main__':
+    foo = [
+        Stars,
+        {"Sirius": [Stars(), Stars]},
+        {"Arcturus": [[{"Vega": Stars}]],
+         "Capella": {"Rigel": {"Procyon": func_1,
```
Comment (dmitry-moroz) [inline solutions/task_1.py]: Make it in single line

File: solutions/task_1.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+class Stars:
+    def __init__(self, star_name=None):
```
Comment (dmitry-moroz) [inline solutions/task_1.py]: Get rid of default None value

**Comment by dmitry-moroz** [comment]
Use flake8 tool to check PEP8

File: solutions/task_1.py (hunk 2 above)
Comment (MorozVladislav) [reply solutions/task_1.py]: It doesn't work without it as when I instantiating Stars() Python is looking for an obligatory argument star_name and crashes each when it fails to find it. On the other hand when I have a default value of star_name I can instantiate Stars() receiving an object with default characteristics.

File: solutions/task_1.py (hunk 1 above)
Comment (MorozVladislav) [reply solutions/task_1.py]: Already done

**Comment by MorozVladislav** [comment]
It seems to me that I've got the very general idea of how all this shit works. Don't be surprised by a great number of stupid commits. I was just practising in git usage. The final commit shall be the right one. One more thing - see my comments on the matter of your corrections.

File: solutions/task_1.py (hunk 2 above)
Comment (dmitry-moroz) [reply solutions/task_1.py]: "Get rid of ..." is not "Remove this code: ', star_name=None' "
I don't want to see unnecessary None here.

File: solutions/task_1.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+class Stars:
```
Comment (dmitry-moroz) [inline solutions/task_1.py]: class Stars(object):

Read about it.

**Comment by dmitry-moroz** [comment]
And one more moment: do not change my code. You have to copy task and modify it without changing task and task code.

I already reverted code changes, so you have to do "**git pull**" before continue make fixes for this PR.
Also I reformatted  'foo' definition, look at this. Is it better than one line code?

File: solutions/task_1.py (hunk 2 above)
Comment (dmitry-moroz) [reply solutions/task_1.py]: ### Read about default keyword arguments!

File: solutions/task_1.py (hunk 1 above)
Comment (dmitry-moroz) [reply solutions/task_1.py]: When I said "Make it in single line" I mean one element of the list per line.
Anyway I fixed it by myself, see comments below.

**Review by dmitry-moroz — CHANGES_REQUESTED** [review]
Oh shit...
[Continue reading about default keyword arguments](http://lmgtfy.com/?iie=1&q=python+%D0%B0%D1%80%D0%B3%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D1%8B+%D0%B7%D0%BD%D0%B0%D1%87%D0%B5%D0%BD%D0%B8%D0%B5+%D0%BF%D0%BE+%D1%83%D0%BC%D0%BE%D0%BB%D1%87%D0%B0%D0%BD%D0%B8%D1%8E)

File: solutions/task_1.py (hunk 3 above)
Comment (dmitry-moroz) [reply solutions/task_1.py]: Why are you ignoring it?

**Review by dmitry-moroz — APPROVED** [review]

**Pull request merged**